cfgen/cfgen

```diff
@@ -476,9 +476,23 @@ def run_command(hostname: str, *args: str) -> str:
                                    timeout=15).decode()
 
 
+def validate_facter(hostname: str) -> None:
+    ver_str = run_command(hostname, 'facter', '--version')
+    # The output may contain commit id like this:
+    # 3.14.8 (commit 4339472f441868ecdae694ffc71e7c8ed0fc24e3)
+    ver_str = ver_str.split(' ')[0]
+
+    if version(ver_str) >= version('3.14.8'):
```

Comment:
Considering https://github.com/Seagate/cortx-hare/pull/1688/files, do we need to make change in the specfile again for sticking only with 3.14.8 and not using ">=", to avoid any surprises that can come with higher versions of facter?

Reply:
We know that versions lower than 3.0.0 work incorrectly sometimes when network interfaces use labels (like `ipaddress_enp175s0f1:c2`). In particular, we [did see this problem](https://jts.seagate.com/browse/EOS-15307?focusedCommentId=1988577&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-1988577) with facter 2.4.1.

As of now, we can't stick to `3.14.8` version in the RPM (note that we stick to puppet-agent package which is not guaranteed to ship this version always). So if our code is too strict, the user will have problems in the future that (s)he will not manage to resolve.

That's why I suggest keeping this more relaxed check.